fix: Release workflow MCPB Bundle job failing - missing @types/node

## Summary

The Release workflow's MCPB Bundle job is failing due to missing Node.js type definitions during TypeScript compilation.

## Error Details

**Workflow**: Release (run #20)
**Job**: MCPB Bundle
**Step**: Install production dependencies / TypeScript compilation

```
src/index.ts(1,21): error TS2307: Cannot find module 'fs' or its corresponding type declarations.
src/index.ts(2,21): error TS2307: Cannot find module 'https' or its corresponding type declarations.
src/index.ts(58,38): error TS2304: Cannot find name 'Buffer'.
src/index.ts(110,9): error TS2304: Cannot find name 'process'.
```

## Root Cause

The MCPB Bundle job installs only production dependencies (`npm ci --omit=dev`), but `@types/node` is listed as a devDependency. TypeScript compilation requires these type definitions to understand Node.js built-in modules like `fs`, `https`, and global objects like `Buffer` and `process`.

## Proposed Solutions

### Option 1: Move @types/node to dependencies (Recommended)
Move `@types/node` from `devDependencies` to `dependencies` in `package.json` since it's required for the production build:

```json
"dependencies": {
  "@types/node": "^22.10.2",
  // ... other deps
}
```

### Option 2: Include dev dependencies in MCPB Bundle
Modify the Release workflow to include dev dependencies during the bundle step, though this may increase bundle size.

### Option 3: Use skipLibCheck
Add `"skipLibCheck": true` to `tsconfig.json`, though this is less ideal as it masks potential type issues.

## Impact

- Release workflow fails after every merge to main
- No new releases can be published until fixed
- This is a pre-existing issue (not introduced by recent PRs)

## Related

- Discovered during post-merge monitoring of PR #40

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/package.json
+++ b/package.json
@@ -72,7 +72,6 @@
   ],
   "devDependencies": {
     "@anthropic-ai/mcpb": "^2.1.0",
-    "@types/node": "^25.0.0",
     "@typescript-eslint/eslint-plugin": "^8.0.0",
     "@typescript-eslint/parser": "^8.0.0",
     "@vitest/coverage-v8": "^4.0.16",
@@ -86,6 +85,7 @@
   },
   "dependencies": {
     "@modelcontextprotocol/sdk": "^1.0.0",
+    "@types/node": "^25.0.0",
     "axios": "^1.7.0",
     "https-proxy-agent": "^7.0.0",
     "jszip": "^3.10.0",

